lib/matplotlib/__init__.py

```diff
@@ -225,7 +225,7 @@ def _is_writable_dir(p):
             t.write(b'1')
         finally:
             t.close()
-    except OSError:
+    except:
```

Comment:
On the other hand there is still code out there that raises strange things (like strings), catching everything here seems like the correct thing to do 'Can we write to this dir yes or no?'  I don't think we really care _why_ we can't write there.